### PR TITLE
:memo: [#308] add Storybook File-Upload component

### LINKED ISSUE #308: Add the file upload component to storybook
We currently don't have all formio components hooked up to storybook, which excludes them from visual regression testing.

1. Add a story file for the file component in `src/formio/components/FileField.stories.mdx`
2. Check the backend form designer for the required JSON for this component
3. Ensure that a URL is set in `component.url` and a mock is set up for this endpoint

The following variants should be set up/documented:

* Happy flow - successful upload
* Error flow - backend error responds with a HTTP 400 with validation errors

The following controls should be available:

* single/multiple upload
* `filePattern` validation (multiple checkboxes, you can find the default list of extensions/mime types here: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/config/constants.py#L60

### PR DESCRIPTION
**Pull request by jiromaykin**
Closes #308

### CODE REVIEW

File: src/formio/components/FileField.stories.mdx
Comment (sergei-maertens) [inline src/formio/components/FileField.stories.mdx:1]: please use `ArgsTable` instead of `Props`

File: src/formio/components/FileField.stories.mdx
Comment (sergei-maertens) [inline src/formio/components/FileField.stories.mdx:10]: please use 'absolute' imports relative to the `src` folder: `mocks/FileUploadMocks`

related: I prefer keeping the mocks and components closely together, so in this case `./FileUpload.mocks.js` is probably a better location.

#369 also adds some more general structure (exposes `BASE_URL` in `story-utils/mocks.js`) so let's wait until that is merged and then rebase this branch on that.

File: src/mocks/FileUploadMocks.js
Comment (sergei-maertens) [inline src/mocks/FileUploadMocks.js:4]: maybe better to just define `const UPLOAD_URL = ${BASE_URL}formio/fileupload`?

File: src/mocks/FileUploadMocks.js
Comment (sergei-maertens) [inline src/mocks/FileUploadMocks.js:16]: please add an example error in the response body - I remember this requires some cruft so you can trigger an invalid upload on a real form and then look at the response data for an example via the browser debugger.

File: src/mocks/FileUploadMocks.js
Comment (sergei-maertens) [inline src/mocks/FileUploadMocks.js:10]: this should be an integer - does not conform to the API docs: http://localhost:8000/api/v2/docs/#tag/formio/operation/formio_fileupload_create

File: src/formio/components/FileField.stories.mdx
Comment (sergei-maertens) [inline src/formio/components/FileField.stories.mdx:108]: this should not be an editable arg - probably we can just bake this directly into the `Template` instead of using an arg for it

File: src/formio/components/FileField.stories.mdx
Comment (sergei-maertens) [inline src/formio/components/FileField.stories.mdx:110]: this maps to the component `key` of the formio component, so I'd use something more appropriate like `fileUpload` here

File: src/formio/components/FileField.stories.mdx
Comment (sergei-maertens) [inline src/formio/components/FileField.stories.mdx:30]: the component itself requires mime types, no? instead of extensions

File: src/formio/components/FileField.stories.mdx
Comment (sergei-maertens) [inline src/formio/components/FileField.stories.mdx:71]: I think this should be hardcoded to `url` and not an arg

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/formio/components/FileField.stories.mdx
Comment (pi-sigma) [reply src/formio/components/FileField.stories.mdx:25]: The component itself takes mime types, not extensions, but the user who creates forms selects mime types only indirectly, viz. by selecting file extensions. If the purpose is purely illustrative, then it doesn't make sense to list mime types.

**Review by sergei-maertens — APPROVED** [review]

**Comment by jiromaykin** [comment]
@sergei-maertens Nice job! I'll take some time to try and understand this one :-D

**Pull request merged**
